subsys/bluetooth/host/hci_core.c

```diff
@@ -3641,6 +3643,50 @@ int bt_enable(bt_ready_cb_t cb)
 	return 0;
 }
 
+int bt_disable(void)
+{
+	int err;
+
+	if (!bt_dev.drv) {
+		BT_ERR("No HCI driver registered");
+		return -ENODEV;
+	}
+
+	if (!bt_dev.drv->close) {
+		return -ENOTSUP;
+	}
+
+	atomic_clear_bit(bt_dev.flags, BT_DEV_ENABLE);
```

Comment:
shouldn't it be done in the end of this function ? Otherwise another thread can call `bt_enable()` while this one is still disabling it ...

Reply:
Makes sense, moved.